internal/gps/source_cache_test.go

```diff
@@ -0,0 +1,473 @@
+		versions := []PairedVersion{
+			NewBranch(br).Pair(rev1),
+			NewVersion(ver).Pair(rev2),
+		}
+		sort.Slice(versions, func(i, j int) bool {
+			return vLess(versions[i], versions[j], true)
+		})
+		c.storeVersionMap(versions, true)
+
+		t.Run("getAllVersions", func(t *testing.T) {
+			got := c.getAllVersions()
+			if len(got) != len(versions) {
+				t.Errorf("expected %d versions but got %d", len(versions), len(got))
+			}
+			sort.Slice(got, func(i, j int) bool {
+				return vLess(got[i], got[j], true)
+			})
+			for i := range versions {
+				if !pairedVersionsEqual(versions[i], got[i]) {
+					t.Errorf("index %d: expected version equal to %v but got %v", i, versions[i], got[i])
+				}
+			}
+		})
+
+		revToUV := map[Revision]UnpairedVersion{
+			rev1: NewBranch(br),
+			rev2: NewVersion(ver),
+		}
+
+		t.Run("getVersionsFor", func(t *testing.T) {
+			for rev, exp := range revToUV {
+				rev, exp := rev, exp
+				t.Run(string(rev), func(t *testing.T) {
+					uvs, ok := c.getVersionsFor(rev)
+					if !ok {
+						t.Errorf("expected version %v but got none", exp)
+					} else if len(uvs) != 1 {
+						t.Errorf("expected one result but got %d", len(uvs))
+					} else {
+						uv := uvs[0]
+						if uv.Type() != exp.Type() {
+							t.Errorf("expected version type %d but got %d", exp.Type(), uv.Type())
+						}
+						if uv.String() != exp.String() {
+							t.Errorf("expected version %s but got %s", exp.String(), uv.String())
+						}
+					}
+				})
+			}
+		})
+
+		t.Run("getRevisionFor", func(t *testing.T) {
+			for exp, uv := range revToUV {
+				exp, uv := exp, uv
+				t.Run(uv.String(), func(t *testing.T) {
+					rev, ok := c.getRevisionFor(uv)
+					if !ok {
+						t.Errorf("expected revision %s but got none", exp)
+					} else if rev != exp {
+						t.Errorf("expected revision %s but got %s", exp, rev)
+					}
+				})
+			}
+		})
+
+		t.Run("toRevision", func(t *testing.T) {
+			for exp, uv := range revToUV {
+				exp, uv := exp, uv
+				t.Run(uv.String(), func(t *testing.T) {
+					rev, ok := c.toRevision(uv)
+					if !ok {
+						t.Errorf("expected revision %s but got none", exp)
+					} else if rev != exp {
+						t.Errorf("expected revision %s but got %s", exp, rev)
+					}
+				})
+			}
+		})
+
+		t.Run("toUnpaired", func(t *testing.T) {
+			for rev, exp := range revToUV {
+				rev, exp := rev, exp
+				t.Run(exp.String(), func(t *testing.T) {
+					uv, ok := c.toUnpaired(rev)
+					if !ok {
+						t.Errorf("expected UnpairedVersion %v but got none", uv)
+					} else if rev != rev {
+						t.Errorf("expected UnpairedVersion %v but got %s", exp, uv)
+					}
+				})
+			}
+		})
+	})
+}
+
+// compareManifests compares two manifests and reports differences as test errors.
+func compareManifests(t *testing.T, exp, got Manifest) {
+	{
+		exp, got := exp.DependencyConstraints(), got.DependencyConstraints()
+		if !projectConstraintsEqual(exp, got) {
+			t.Errorf("expected constraints equal to %v but got %v", exp, got)
```

Comment:
there's a `(WNT):`, `(GOT):` pattern used elsewhere in gps' tests. For consistency's sake, let's continue the pattern.